test/parser.test.js

```diff
@@ -383,31 +383,424 @@ QUnit.module('m3u8s', function(hooks) {
     );
   });
 
-  QUnit.module('warn/info');
+  QUnit.module('warn/info', {
+    beforeEach() {
+      this.warnings = [];
+      this.infos = [];
 
-  QUnit.test('triggers warning for missing EXT-X-START TIME-OFFSET attribute', function(assert) {
+      this.parser.on('warn', (warn) => this.warnings.push(warn.message));
+      this.parser.on('info', (info) => this.infos.push(info.message));
 
-    const manifest = [
+    }
+  });
+  QUnit.test('warn when #EXT-X-TOTAL-DURATION is invalid', function(assert) {
+    this.parser.push([
+      '#EXT-X-VERSION:3',
+      '#EXT-X-MEDIA-SEQUENCE:0',
+      '#EXT-X-DISCONTINUITY-SEQUENCE:0',
+      '#EXT-X-TARGETDURATION:foo',
+      '#EXTINF:10,',
+      'media-00001.ts',
+      '#EXT-X-ENDLIST'
+    ].join('\n'));
+    this.parser.end();
+
+    const warnings = [
+      'ignoring invalid target duration: undefined'
```

Comment:
this has `undefined` because we tried to parse `foo` as an int which failed, and we just didn't set a value in that case?

If we would want to fix this, we'd want to do it in a separate PR.

Reply:
I think I will be fixing up some of the old warnings and info logs to better represent what we are doing now in another pull request.